src/legacy/core_plugins/vis_type_timeseries/server/routes/vis.js

```diff
@@ -18,13 +18,40 @@
  */
 
 import { getVisData } from '../lib/get_vis_data';
+import { visPayloadSchema } from './post_vis_schema';
 import Boom from 'boom';
+import { i18n } from '@kbn/i18n';
 
-export const visDataRoutes = server => {
+export const visDataRoutes = (server, { logFailedValidation }) => {
   server.route({
     path: '/api/metrics/vis/data',
     method: 'POST',
     handler: async req => {
+      const { error } = visPayloadSchema.validate(req.payload);
```

Comment:
Which method is preferred for checking payload? 
I see some routes use the following way for validataion
![image](https://user-images.githubusercontent.com/20072247/71177172-6db3a400-227c-11ea-83cc-825b84d1e121.png)

@timroes  @lukeelmers  @flash1293

Reply:
We can't do the validaton as part of `validate` because we want to check the flag whether or not we should fail the request when the validation reports an error. Switching to kbn schema makes sense, it doesn't look like features exclusive to Joi are used